sklearn/cluster/tests/test_dbscan.py

```diff
@@ -192,6 +192,15 @@ def test_dbscan_metric_params():
     assert_array_equal(core_sample_1, core_sample_3)
     assert_array_equal(labels_1, labels_3)
 
+    with pytest.warns(SyntaxWarning):
+        # Test that checks p is ignored in favor of metric_params={'p': <val>}
+        db = DBSCAN(metric='minkowski', metric_params={'p': p}, eps=eps, p=p+1,
+                    min_samples=min_samples, algorithm='ball_tree').fit(X)
+        core_sample_4, labels_4 = db.core_sample_indices_, db.labels_
+
+    assert_array_equal(core_sample_1, core_sample_4)
+    assert_array_equal(labels_1, labels_4)
+
```

Comment:
should also test that no warnings are raised if `p=None` the same way as it's done here: https://github.com/scikit-learn/scikit-learn/blob/670b85c9e9cec05210e8596bc1fb9ca66787162f/sklearn/compose/tests/test_column_transformer.py#L1240-L1243